shopelectro/selenium/elements/cart.py

```diff
@@ -22,20 +25,33 @@ def _hover(self):
             (By.CLASS_NAME, 'js-cart-wrapper')
         ))
 
+    @contextmanager
+    def wait_changes(self):
+        def wait_changes(browser):
+            try:
+                return positions_before != self.positions()
```

Comment:
try-except looks dangerous. Lot's look at the cart clearing case:
- we wrap non-empty cart clearing with this context manager
- after clearing positions_before is not empty yet
- but self.positions() are already empty
- so `positions_before != self.positions()` should be `True`
- but with this try-except wrapping the function will return `False`

Try to wrap self..clear() with the context manager to check my assumption.

You should catch the exception only for `self.positions()`, not the whole expression `return positions_before != self.positions()`.

Reply:
>so positions_before != self.positions() should be True
but with this try-except wrapping the function will return False

`self.positions()` checks cart emptiness, so it dosn't need this try/catch, but it is needed for cart's position hash calculation.  

I'll create todo to clarify the point

>Try to wrap self..clear() with the context manager to check my assumption.

We already have a `remove` method that behaves the same when positions quantity = 1.  We already have used it in tests